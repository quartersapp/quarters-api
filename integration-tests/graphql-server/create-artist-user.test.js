/* eslint-env mocha */

const request = require('supertest')
const app = require('lib/app').listen()
const { User, City } = require('lib/db/models')
const { truncate, enableSnapshots } = require('test-helpers')

describe('integration-tests/graphql-server/create-artist-user', () => {
  beforeEach(() => truncate([User, City]))
  beforeEach(enableSnapshots)

  it('can create a new artist/user', async () => {
    const city = await City.create({
      name: 'Toronto',
      googlePlaceId: 'place-id'
    })

    const variables = {
      input: {
        firstName: 'Thomas',
        lastName: 'Dashney',
        artistName: 'Lost Cousins',
        email: 'test@example.com',
        password: 'pass1234',
        cityId: city.id
      }
    }

    const query = `
      mutation CreateArtistUser($input: CreateArtistUserInput!) {
        createArtistUser(input: $input) {
          artist {
            name,
            city {
              name
            }
          }
        }
      }
    `

    const { body } = await request(app)
      .post('/graphql')
      .send({ query, variables })

    expect(body).to.matchSnapshot()
  })
})
