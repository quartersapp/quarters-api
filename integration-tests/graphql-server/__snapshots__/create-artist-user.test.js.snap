// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can create a new artist/user 1`] = `
Object {
  "data": Object {
    "createArtistUser": Object {
      "artist": Object {
        "city": Object {
          "name": "Toronto",
        },
        "name": "Lost Cousins",
      },
    },
  },
}
`;
